components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/APIMgtResponseHandler.java

```diff
@@ -215,7 +215,13 @@ public boolean mediate(MessageContext mc) {
             stream.setRequestTimestamp(
                     Long.parseLong((String) mc.getProperty(APIMgtGatewayConstants.REQUEST_START_TIME)));
             stream.setResponseCacheHit(cacheHit);
-            stream.setResponseCode((Integer) axis2MC.getProperty(SynapseConstants.HTTP_SC));
+            int responseCode;
+            if (axis2MC.getProperty(SynapseConstants.HTTP_SC) instanceof String) {
```

Comment:
This becomes a string when it's modified via a mediation like follows.

   <property name="HTTP_SC" value="200" scope="axis2"/>

Otherwise, the status set to the message context is an int.